index.js

```diff
@@ -698,49 +1348,102 @@ function find_comment_dfver(b)
 					var vers = m[1].split('-');
 					var min = vers[0] || 0;
 					var max = vers[1] || 9999;
-					
+
 					return (dfver >= min && dfver <= max);
 				}
-					
+
 				break;
 			}
 		}
 	}
-	
+
 	return true;
-}	
+}
 
-function process(body, ctx) {
+function processAST(body, ctx) {
 	var rettype = null;
 
 	body.forEach(function(b) {
 		if (b.type == 'FunctionDeclaration') {
-			var c = b.isLocal ? ctx : rootctx;
+			var c = b.isLocal ? ctx : ctxstack[ctxstack.length - 1];
+			var n = flatten(b.identifier, ctx);
+			var existing = expandtype(n, ctx, b.loc.start.line, true);
 			c.functions = c.functions || {};
-			c.functions[b.identifier.name] = b;
-			c.functions[b.identifier.name]._src = srcstack[srcstack.length-1];
-			c.functions[b.identifier.name]._ctx = ctx;
-			c.types[b.identifier.name] = { _type:'function', _node:b, _ctx:ctx, _src:srcstack[srcstack.length-1] }
-			
-			if (!b.isLocal)
-				checked_global_fns[b.identifier.name] = false;
-			
+			c.functions[n] = b;
+			c.functions[n]._src = srcstack[srcstack.length-1];
+			c.functions[n]._ctx = ctx;
+			c.types[n] = { _type:'function', _node:b, _ctx:ctx, _src:srcstack[srcstack.length-1] }
+			var base = null;
+			var member = null;
+			if (b.identifier.type == 'MemberExpression') {
+				base = checktype(b.identifier.base, ctx);
+				member = b.identifier.identifier.name;
+				if (base && base._type) {
+					base._methods = base._methods || {};
+					base._methods[member] = c.types[n];
+				}
+			}
+			var ctypes = c.types;
+
+			if (!b.isLocal && srcstack.length == 1) {
+				unchecked_global_fns[n] = b.loc.start.line;
+			}
+
+			var found = false;
+			if (base && base._defclass && member) {
+				for (var p = base; p; p = p._super) {
+					if (p._type == 'Screen' && (member == 'onShow' || member == 'onDismiss' || member == 'onDestroy' || member == 'onRender' || member == 'onIdle' || member == 'onHelp' || member == 'onGetSelectedUnit' || member == 'onGetSelectedItem' || member == 'onGetSelectedJob' || member == 'onGetSelectedBuilding')) {
```

Comment:
And this ideally shouldn't be hardcoded right in process() either.

Reply:
I have an `anyfunc:none` type that would work for this one, and the others can do similar things. I can also remove the `__gui_Painter` type alias.